cli/src/compute_budget.rs

```diff
@@ -20,6 +20,35 @@ pub(crate) enum UpdateComputeUnitLimitResult {
     NoInstructionFound,
 }
 
+// Returns the compute unit limit used during simulation
+pub(crate) fn simulate_for_compute_unit_limit(
```

Comment:
I feel like this function should also check whether message has a compute limit instruction. Otherwise it could be misused because if you add the compute limit instruction after simulation, it would increase the compute units consumed.

Reply:
That makes sense, I've refactored a bit to do that -- let me know what you think